services/web/server/src/simcore_service_webserver/meta_models_repos.py

```diff
@@ -0,0 +1,84 @@
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Tuple, Union
+from uuid import UUID
+
+from aiopg.sa.result import RowProxy
+from models_library.basic_types import SHA1Str
+from models_library.projects_nodes import Node
+from pydantic import BaseModel, PositiveInt, StrictBool, StrictFloat, StrictInt
+from pydantic.networks import HttpUrl
+
+BuiltinTypes = Union[StrictBool, StrictInt, StrictFloat, str]
+
+# alias for readability
+# SEE https://pydantic-docs.helpmanual.io/usage/models/#orm-mode-aka-arbitrary-class-instances
+
+CommitProxy = RowProxy
+TagProxy = RowProxy
+CommitLog = Tuple[CommitProxy, List[TagProxy]]
+
+ProjectProxy = RowProxy
+ProjectDict = Dict
+
+HEAD = f"{__file__}/ref/HEAD"
+
+RefID = Union[int, str]
+
+
+class Checkpoint(BaseModel):
+    id: PositiveInt
+    checksum: SHA1Str
+    created_at: datetime
+    tags: Tuple[str, ...]
+
+    message: Optional[str] = None
+    parents_ids: Tuple[PositiveInt, ...] = None  # type: ignore
+
+    @classmethod
+    def from_commit_log(cls, commit: RowProxy, tags: List[RowProxy]) -> "Checkpoint":
+        return cls(
+            id=commit.id,
+            checksum=commit.snapshot_checksum,
+            tags=tuple(tag.name for tag in tags),
+            message=commit.message,
+            parents_ids=(commit.parent_commit_id,) if commit.parent_commit_id else None,
+            created_at=commit.created,
+        )
+
+
+class WorkbenchView(BaseModel):
+    """A view (i.e. read-only and visual) of the project's workbench"""
+
+    class Config:
+        orm_mode = True
+
+    workbench: Dict[UUID, Node] = {}
+    ui: Dict[UUID, Any] = {}
```

Comment:
same here

Reply:
iti s the same, right? :-D

Reply:
yes it is the same. but aliases are there in the case they would change and to make it more understandable. Here you actually *know* these are the same. but for someone not knowing the code is a bit more difficult.